tests/lib/default.nix

```diff
@@ -0,0 +1,62 @@
+{ pkgs }:
+rec {
```

Comment:
Every part of this attribute will be the public testing API as soon as it gets merged and will have to be supported for probably all eternity.
Big -1 from me for the current exposed symbols, there needs to be a lot more consensus first, and new (non-internal) symbols should only be introduced with care and properly documented.

Reply:
As I said, abstraction is currently done wrong, because there are not enough tests to generalise from. The comment-gathering re: what _kinds_ of tests can be done seems to happen somehow.

Consensus won't happen anyway, but right now even consistency will be a step ahead.

What I did miss is that I haven't added `[RFC]` and `[WIP]` tags in the name to clarify the intent; fixed that.